### PR TITLE
Api routes. With supabase

### PR DESCRIPTION
**Pull request by jenilparmar**

### CODE REVIEW

**Comment by jenilparmar** [comment]
starting of work

File: src/app/pages/api/v1/blogs.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    let query = supabase
+      .from('blogs')
+      .select('*')
+      .order('created_at', { ascending: false });
+
+    if (limit) {
+      const limitNum = parseInt(limit, 10);
+      query = query.limit(limitNum);
+
+      if (page) {
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/blogs.js:17]: it's not compulsory to use page with limit, one can use page without limit too... and as far as I can understand this, it's only considering page if limit is used

File: src/app/pages/api/v1/create/event.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    try {
+      const { data, error } = await supabase
+        .from('events')
+        .insert([event]);
+
+      if (error) {
+        return res.status(500).json({ success: false, message: error.message });
+      }
+      return res.status(200).json({ success: true, message: 'Event created successfully', id: data[0].id });
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/create/event.js:25]: I don't think supabase.from.insert statement return the row id without using a .select() on it??

File: src/app/pages/api/v1/events.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  if (req.method === 'GET') {
+    const { limit, page, category } = req.query;
+
+    let query = supabase
+      .from('events')
+      .select('*')
+      .order('date', { ascending: false });
+
+    if (category) {
+      query = query.eq('category', category);
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/events.js:14]: there's no `category` column in the table... it was to be calculated based on the date

File: src/app/pages/api/v1/events.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    if (limit) {
+      const limitNum = parseInt(limit, 10);
+      query = query.limit(limitNum);
+
+      if (page) {
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/events.js:21]: same thing here (as was in /blogs endpoint), the page can be used without limit too...

File: src/app/pages/api/v1/get/blog.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    if (!id) {
+      return res.status(400).json({ success: false, message: 'Missing blog ID' });
+    }
+
+    try {
+      const { data, error } = await supabase
+        .from('blogs')
+        .select('*')
+        .eq('id', id)
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/get/blog.js:16]: as per docs, you'll get Title of blog + blog ID in the id parameter, not just the id.

File: src/app/pages/api/v1/submit/blog.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    if (!title || !blog) {
+      return res.status(400).json({ success: false, message: 'Missing required fields: title or blog content' });
+    }
+
+    try {
+      const { data, error } = await supabase
+        .from('blogs')
+        .insert([{ title, intro, assets, blog }]);
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/submit/blog.js:20]: `blog` is not a column in the table (please have a look on the schema). Blog has to be saved in the S3 bucket and then an entry should be added in the assets object, with "blog": <whatever file name is> 

Also, you have to fetch writer's user id from the auth token and insert writer id in the table too

File: src/app/pages/api/v1/create/event.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+// pages/api/v1/create/event.js
+import { supabase } from '../../../../supabaseClient';
+
+export default async function handler(req, res) {
+  if (req.method === 'POST') {
+    const authHeader = req.headers.authorization;
+    const { event } = req.body;
+
+    if (!authHeader || !authHeader.startsWith('Bearer ')) {
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/create/event.js:9]: is just having a Bearer token enough? Aren't you gonna verify the session?

File: src/app/pages/api/v1/submit/blog.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+// pages/api/v1/submit/blog.js
+import { supabase } from '../../../../supabaseClient';
+
+export default async function handler(req, res) {
+  if (req.method === 'POST') {
+    const authHeader = req.headers.authorization;
+    const { title, intro, assets, blog } = req.body;
+
+    if (!authHeader || !authHeader.startsWith('Bearer ')) {
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/submit/blog.js:9]: is just having a Bearer token enough? Aren't you gonna verify the session?

File: src/app/pages/api/v2/delete/event.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+// pages/api/v1/delete/event.js
+import { supabase } from '../../../../supabaseClient';
+
+export default async function handler(req, res) {
+  if (req.method === 'DELETE') {
+    const { id } = req.query;
+    const authHeader = req.headers.authorization;
+
+    if (!authHeader || !authHeader.startsWith('Bearer ')) {
```
Comment (thedevyashsaini) [inline src/app/pages/api/v2/delete/event.js:9]: is just having a Bearer token enough? Aren't you gonna verify the session?

File: src/app/pages/api/v1/event.js
Comment (thedevyashsaini) [inline src/app/pages/api/v1/event.js:1]: the endpoint should be "/api/v1/get/event" as per docs, not "/api/v1/event"

File: src/app/pages/api/v1/create/event.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      return res.status(401).json({ success: false, message: 'Missing or invalid authorization header' });
+    }
+
+    if (!event || !event.name || !event.date) {
+      return res.status(400).json({ success: false, message: 'Missing required event fields' });
+    }
+
+    try {
+      const { data, error } = await supabase
```
Comment (thedevyashsaini) [inline src/app/pages/api/v1/create/event.js:18]: Only admins should be able to add a new event, right? For that RLS policies are written but how would they know who's making the request? You have to set Supabase client to use the user's JWT

File: src/app/pages/api/v2/delete/event.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+      return res.status(401).json({ success: false, message: 'Missing or invalid authorization header' });
+    }
+
+    if (!id || id.length !== 36) {
+      return res.status(400).json({ success: false, message: 'Invalid or missing event ID' });
+    }
+
+    try {
+      const { data, error } = await supabase
```
Comment (thedevyashsaini) [inline src/app/pages/api/v2/delete/event.js:18]: only admins should be able to delete an event, right? For that RLS policies are written but how would they know who's making the request? You have to set Supabase client to use the user's JWT

**Review by thedevyashsaini — CHANGES_REQUESTED** [review]
I've reviewed the commit and left comments on the areas that need attention. Please address these issues, and feel free to let me know if you disagree with any of my points.

**Pull request closed without merging**